### ISSUE TITLE
CLI: Show help text if a nonexistent subcommand is given

### ISSUE THREAD

**Issue by parnikkapore** [jackfirth/resyntax#140]
Currently, if someone types:
```sh
resyntax my-file.rkt
```
The result will be an error message that would be very confusing to the user:
```
match: no matching clause for "my-file.rkt"
  context...:
   .../racket/cmdline.rkt:191:51
   body of (submod "[...]resyntax/cli.rkt" main)
```

I think we should show a help text listing the available subcommands here.

**Comment by jackfirth** [comment]
Yup, this would definitely be a good improvement. The `command-line` API makes it kind of tricky to have subcommands, so the current implementation is a bit of a hack with very little polish.

**Comment by parnikkapore** [comment]
I actually went searching yesterday for a subcommand package, but I didn't find any 🤷

**Comment by Metaxal** [comment]
The docs also mention the `resyntax` executable, but it doesn't seem it exists anywhere, and it took me some search and code reading to find out that I should run `racket -l- resyntax analyze --help` to start.

**Comment by jackfirth** [comment]
> The docs also mention the `resyntax` executable, but it doesn't seem it exists anywhere, and it took me some search and code reading to find out that I should run `racket -l- resyntax analyze --help` to start.

It exists, but unfortunately the default instructions to add racket to your `$PATH` only adds paths to launchers installed by _installation scope_ packages, not _user scope_ ones, see [this comment](https://github.com/racket/racket/issues/3708#issuecomment-790023077). You can make it actually show up by installing resyntax in installation scope, or by changing your `$PATH` to include wherever user-scope launchers go.

**Comment by jackfirth** [comment]
I should probably put that info in the docs...

**Comment by Metaxal** [comment]
That's why `racket -l- resyntax` is a more robust command. In particular because people using resyntax will necessarily have racket installed too, so it's not like resyntax is a standalone program anyway. A shortcut is nice, but since it's fiddly for now, I would advise to go for the most robust solution, in particular in development stages, to smooth out the user experience of trying resyntax out.

**Comment by Metaxal** [comment]
> [...] or by changing your $PATH to include wherever user-scope launchers go.

I'm (likely) not going to do that, because the paths are depending on the racket version, rather than going in something like `~/.local/bin`, and I don't want to have to maintain my $PATH each time I install a new Racket snapshot (which is often enough).

**Comment by jackfirth** [comment]
> That's why `racket -l- resyntax` is a more robust command. In particular because people using resyntax will necessarily have racket installed too, so it's not like resyntax is a standalone program anyway. A shortcut is nice, but since it's fiddly for now, I would advise to go for the most robust solution, in particular in development stages, to smooth out the user experience of trying resyntax out.

Just checking, does `racket -l- resyntax` work or is it `racket -l- resyntax/cli`? I think it should be the latter but if the former is working for you there might be some magic in how the launchers are set up that I'm not understanding.

**Comment by Metaxal** [comment]
`racket -l- resyntax` doesn't work by default. But in the PR I added this line:
https://github.com/jackfirth/resyntax/pull/147/files#diff-9d629e9198e03ca0a3046f8f44e650d4f7e3b650ca2807ec982977b3c1c34797R103
which tells the user to try `racket -l- resyntax/clip` instead.

**Comment by jackfirth** [comment]
Closed in #208.
